pyomo/solvers/tests/piecewise_linear/test_examples.py

```diff
@@ -39,57 +40,74 @@ def run_convert2lp(self, name):
     def test_step_lp(self):
         """Test examples/pyomo/piecewise/step.py"""
         self.run_convert2lp('step.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'step.lp', tolerance=_LP_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.lp'), currdir+'step.lp'))
 
     def test_step_nl(self):
         """Test examples/pyomo/piecewise/step.py"""
         self.run_convert2nl('step.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'step.nl', tolerance=_NL_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.nl'), currdir+'step.nl'))
         os.remove(join(currdir,'unknown.row'))
         os.remove(join(currdir,'unknown.col'))
 
     def test_nonconvex_lp(self):
         """Test examples/pyomo/piecewise/nonconvex.py"""
         self.run_convert2lp('nonconvex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'nonconvex.lp', tolerance=_LP_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.lp'), currdir+'nonconvex.lp'))
 
     def test_nonconvex_nl(self):
         """Test examples/pyomo/piecewise/nonconvex.py"""
         self.run_convert2nl('nonconvex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'nonconvex.nl', tolerance=_NL_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.nl'), currdir+'nonconvex.nl'))
         os.remove(join(currdir,'unknown.row'))
         os.remove(join(currdir,'unknown.col'))
 
     def test_convex_lp(self):
         """Test examples/pyomo/piecewise/convex.py"""
         self.run_convert2lp('convex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'convex.lp', tolerance=_LP_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.lp'), currdir+'convex.lp'))
 
     def test_convex_nl(self):
         """Test examples/pyomo/piecewise/convex.py"""
         self.run_convert2nl('convex.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.nl'), currdir+'convex.nl', tolerance=_NL_diff_tol)
+        self.assertTrue(cmp(join(currdir,'unknown.nl'), currdir+'convex.nl'))
         os.remove(join(currdir,'unknown.row'))
         os.remove(join(currdir,'unknown.col'))
 
     def test_indexed_lp(self):
         """Test examples/pyomo/piecewise/indexed.py"""
         self.run_convert2lp('indexed.py')
-        self.assertFileEqualsBaseline(join(currdir,'unknown.lp'), currdir+'indexed.lp', tolerance=_LP_diff_tol)
+        with open(join(currdir,'unknown.lp'), 'r') as f1, \
+            open(currdir+'indexed.lp', 'r') as f2:
+                f1_contents = list(filter(None, f1.read().split()))
+                f2_contents = list(filter(None, f2.read().split()))
+                for item1, item2 in zip(f1_contents, f2_contents):
+                    try:
+                        self.assertEqual(item1, item2)
+                    except:
+                        self.assertAlmostEqual(float(item1), float(item2))
```

Comment:
As mentioned before, I would recommend reversing the order of these asserts (try the float, fall back on the assertEqual).